model/deliveryLog/listener/DeliveryLogTimerAdjustedEventListener.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020  (original work) Open Assessment Technologies SA;
+ *
+ * @author Oleksandr Zagovorychev <zagovorichev@gmail.com>
+ */
+declare(strict_types=1);
+
+namespace oat\taoProctoring\model\deliveryLog\listener;
+
+use common_Exception;
+use common_exception_Error;
+use common_exception_NotFound;
+use common_ext_ExtensionException;
+use Context;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\taoProctoring\model\deliveryLog\DeliveryLog;
+use oat\taoProctoring\model\deliveryLog\event\DeliveryLogEvent;
+use oat\taoProctoring\model\event\DeliveryExecutionTimerAdjusted;
+use oat\taoProctoring\model\implementation\TestSessionService;
+use oat\taoQtiTest\models\QtiTestExtractionFailedException;
+
+class DeliveryLogTimerAdjustedEventListener extends ConfigurableService
+{
+    /**
+     * @param DeliveryExecutionTimerAdjusted $event
+     * @throws InvalidServiceManagerException
+     * @throws QtiTestExtractionFailedException
+     * @throws common_Exception
+     * @throws common_exception_Error
+     * @throws common_exception_NotFound
+     * @throws common_ext_ExtensionException
+     */
+    public function onTimerAdjusted(DeliveryExecutionTimerAdjusted $event): void
```

Comment:
I would suggest to rename this method after **what** does it do instead of **when**. We already know that it happens on timer adjusted event based on input parameter event type. If we use more informative name what does it do when this event happens it will be more clear.
As another possible improvement we could rename method to something more generic like `DeliveryStateChangeEventListeener`. Then in future we can move other delivery logging functionality (authorized, paused, terminated, etc.) here. We already have all (or almost all) required dependencies here.

Reply:
I think that class name should be different, it makes easier to add or replace listeners in other extensions (instead of methods rewriting)

Reply:
onTimerAdjusted -> adjustTime